test/com/google/javascript/jscomp/Es6VariableReferenceCheckTest.java

```diff
@@ -277,6 +277,10 @@ public void testClass() {
     assertNoWarning("class A { f() { return 1729; } }");
   }
 
+  public void testRedeclareClassName() {
+    assertNoWarning("var Clazz = class Foo {}; var Foo = 3;");
```

Comment:
@Dominator008 I would propose to just ban the pattern of using two different names for the same class. What's a good reason to support this?